packages/core/src/storybook/components/toc/TableOfContents.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect, useCallback } from "react";
+import styles from "./TableOfContents.module.scss";
+import Flex from "../../../components/Flex/Flex";
+import { Timeline } from "@vibe/icons";
+import IconButton from "../../../components/IconButton/IconButton";
+import cx from "classnames";
+
+export default function TableOfContents() {
+  const [isOpen, setIsOpen] = useState(false);
+  const [headers, setHeaders] = useState([]);
+  const [activeHeader, setActiveHeader] = useState("");
+
+  const handleMouseEnter = () => setIsOpen(true);
+  const handleMouseLeave = () => setIsOpen(false);
+
+  useEffect(() => {
+    const contentElement = document.querySelector("#storybook-docs");
+    const foundHeaders = Array.from(contentElement.getElementsByTagName("h2"))
+      .filter((h: HTMLElement) => {
+        return Boolean(h.id);
+      })
+      .map(h => {
+        return { id: h.id, text: h.textContent || "", element: h };
+      })
+      // limit to 15 headers
+      .slice(0, 15);
+    setHeaders(foundHeaders);
+  }, [isOpen]);
```

Comment:
I'm not sure I understand if the filtering got in
is it? or do we include inner h2s from stories?